pkg/etcdenvvar/envvarcontroller.go

```diff
@@ -59,9 +60,11 @@ func NewEnvVarController(
 	infrastructureInformer configv1informers.InfrastructureInformer,
 	networkInformer configv1informers.NetworkInformer,
 	eventRecorder events.Recorder,
+	etcdSpec operatorv1.EtcdSpec,
```

Comment:
that spec can change at runtime, don't you want to use the client/informer instead?

Reply:
I'd still use the informer instead (and wire it up in the controller factory), because that way you can have the controller sync with changes. When you do a GET with the client you're at the mercy of some other resource changing at the same time.